Sources/MapboxMaps/Foundation/Camera/CameraAnimator.swift

```diff
@@ -1,20 +1,40 @@
 import UIKit
+import CoreLocation
 
-// MARK: CameraAnimator Class
-public class CameraAnimator: NSObject {
+@objc public protocol CameraAnimatorProtocol: AnyObject {
+
+    /// Stops the animation in its tracks and calls any provided completion
+    func stopAnimation()
+
+    /// The current state of the animation
+    var state: UIViewAnimatingState { get }
+
+    /// TODO: Move this to a non-public interface
+    func update()
+}
 
-    // MARK: Stored Properties
+public typealias CameraAnimation = (inout CameraTransition) -> Void
```

Comment:
Let's see if we can think of a better name for this typealias. As it stands, it makes me think that it's modeling some in-flight representation of the animation. Maybe `CameraTransitionProvider`?

Reply:
Do we even need a typealias?

Reply:
Perhaps not.. I'll get rid of it for now. We can introduce it later if needed.